### PR TITLE
Handle failed of iceconnectionstate and signalingstate

### PR DESCRIPTION
**Pull request by vogdb**

### CODE REVIEW

**Comment by vogdb** [comment]
@zazabe wdyt about this?

**Comment by zazabe** [comment]
- could we rather use `addEventListener`?
- maybe we should not care about [`disconnected`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/iceConnectionState#RTCIceConnectionState_enum):    
"This is a less stringent test than "failed" and may trigger intermittently and resolve just as spontaneously on less reliable networks."
- watch the `signalingState` seems good to me, could be useful later to catch some wrong state as [described in the MDN doc](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/signalingState):     
  "For example, if you receive an answer while the signalingState isn't "have-local-offer", you know that something is wrong."

**Comment by zazabe** [comment]
In case of a failure, we should throw also i think...

**Comment by vogdb** [comment]
> - maybe we should not care about [`disconnected`]...

ok, I just took this handler from https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API/Signaling_and_video_calling#Dealing_with_state_changes

**Comment by zazabe** [comment]
ok, let's keep this `disconnected` if it's recommended by Mozilla ;)

**Comment by vogdb** [comment]
@zazabe please review. I've tested it. Works fine.

File: src/webrtc/media-plugin.js [hunk 1]
```diff
@@ -181,4 +169,59 @@ MediaPlugin.prototype._onTrickle = function(incomeMessage) {
   }.bind(this));
 };
 
+MediaPlugin.prototype.closePeerConnection = function() {
+  ['track', 'addstream', 'icecandidate', 'signalingstatechange', 'iceconnectionstatechange']
```
Comment (zazabe) [inline src/webrtc/media-plugin.js]: why not iterate over all `_pcListeners` properties instead?

**Comment by vogdb** [comment]
@zazabe please review

**Comment by zazabe** [comment]
lgtm!

**Pull request merged**